core/server/services/slack.js

```diff
@@ -48,11 +54,51 @@ function ping(post) {
             return;
         }
 
+        var blogUrl = config.get('url');
+        var blogTitle = settingsCache.get('title');
+
         slackData = {
-            text: message,
+            // We are handling the case of test notification here by checking
+            // if it is a post or a test message to check webhook working.
+            text: schema.isPost(post) ? `Notification from *${blogTitle}* :ghost:` : message,
             unfurl_links: true,
             icon_url: imageLib.blogIcon.getIconUrl(true),
-            username: 'Ghost'
+            username: 'Ghost',
+            // We don't want to send attachment if it is a test notification.
+            attachments: schema.isPost(post) ? [
+                {
+                    fallback: 'Sorry, content cannot be shown.',
+                    title: title,
+                    title_link: message,
+                    author_name: blogTitle,
+                    image_url: post && post.feature_image ? `${blogUrl}${post.feature_image}` : null,
+                    color: '#008952',
+                    fields: [
+                        {
+                            title: 'Description',
+                            value: post.html ? `${post.html.replace(/<[^>]+>/g, '').split('.').slice(0, 3).join('.')}.` : '',
+                            short: false
+                        }
+                    ]
+                },
+                {
+                    fallback: 'Sorry, content cannot be shown.',
+                    color: '#008952',
+                    thumb_url: author && author.profile_image ? `${blogUrl}${author.profile_image}` : null,
+                    fields: [
+                        {
+                            title: 'Author',
+                            value: author && author.slug && author.name ? `<${blogUrl}/author/${author.slug} | ${author.name}>` : null,
```

Comment:
The author taxonomy url is not static, it can be modified in the routes.yaml, [see](https://docs.ghost.org/docs/dynamic-routing).
Please use `urlService.getUrlByResourceId(author.id, {absolute:true})`.

Reply:
And FYI: there is always one author.